src/dmclock_server.h

```diff
@@ -72,6 +72,12 @@ namespace crimson {
       std::numeric_limits<double>::lowest();
     constexpr uint tag_modulo = 1000000;
 
+    // we're abstracting cost to its own type to better allow for
+    // future changes; we're assuming that Cost is relatively small
+    // and that it would be more efficient to pass-by-value than
+    // by-reference.
+    using Cost = uint64_t;
```

Comment:
@tchaikov's question regarding whether delta and rho should be uint64_t's raises a similar question w.r.t. Cost. Should Cost also be uint32_t? If we use cost=1 as a base, would there be an op that had a relative cost of more than 4 billion? If not, uint32_t would be sufficient.

@cbodley @tchaikov Your thoughts?

Reply:
i think 4G is a very large number in the context of a networking protocol at this moment. we can cap the cost at `std::numeric_limit<uint32>::max()` (which is 4G). if its value is actually greater than it.

Reply:
you've convinced me that uint32_t is sufficient here